src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.Unix.cs

```diff
@@ -0,0 +1,156 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Diagnostics;
+
+namespace System.Formats.Tar
+{
+    internal static partial class TarHelpers
+    {
+        private static readonly Lazy<UnixFileMode> s_umask = new Lazy<UnixFileMode>(DetermineUMask);
+
+        private static UnixFileMode DetermineUMask()
+        {
+            Debug.Assert(!OperatingSystem.IsWindows());
```

Comment:
A poor man's _get umask_ implementation.

We can't call [umask](https://man7.org/linux/man-pages/man2/umask.2.html) since that changes the umask while getting it.

On Linux, the umask can be read from the `/proc/<pid>/status` file.